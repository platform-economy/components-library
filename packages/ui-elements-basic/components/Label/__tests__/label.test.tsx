import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';

/*
A set of utilities for testing Styled Components with Jest.
This package improves the snapshot testing experience and
provides a brand new matcher to make expectations on the style rules.
*/
import 'jest-styled-components';

jest.mock('react-i18next', () => ({
  // tslint:disable-next-line:no-any
  NamespacesConsumer: (props: any) => (
    <div id="NamespacesConsumer">{props.children((text: string) => text)}</div>
  ),
}));

// Component to test
import { Label } from '../index';

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('Label', () => {
      it('should render with default properties only', () => {
        const tree = TestRenderer
          .create(<Label title="Test Title" />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
