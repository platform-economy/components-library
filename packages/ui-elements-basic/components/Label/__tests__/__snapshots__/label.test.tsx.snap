// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components Label should render with default properties only 1`] = `
.c0 {
  display: inline;
  color: #00aaff;
  font-size: .938em;
  font-weight: bold;
  padding: 6px 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .staticText {
  color: #000;
}

<div
  id="NamespacesConsumer"
>
  <label
    className="c0"
  >
    <span
      className="staticText"
    >
      staticLabel
      : 
    </span>
    Test Title
  </label>
</div>
`;
